---
permalink: /publication/
title: "Papers & Preprints"
layout: splash
---
## Publication

**See [google scholar](https://scholar.google.com/citations?hl=en&user=Y3Cqt0cAAAAJ&view_op=list_works&sortby=pubdate) for the full publication list.** 

[Citations](https://raw.githubusercontent.com/xulabs/xulabs.github.io/master/doc/reference_xulab.bib) in latex bib format. The source code in [AITom](https://github.com/xulabs/aitom) [publication list](https://github.com/xulabs/aitom/blob/master/doc/publications.md).

### Selected publications

1. [ET] Zhao et al. CryoSAM: Training-free CryoET Tomogram Segmentation with Foundation Models. Medical Image Computing & Computer Assisted Intervention (MICCAI) 2024.
1. [AS, CV] Deep Video Anomaly Detection in Automated Laboratory Setting. SSRN Preprint. [doi:10.2139/ssrn.4887151](http://dx.doi.org/10.2139/ssrn.4887151)
1. [AS,CV] Jain et al. Cross-Domain Learning for Video Anomaly Detection with Limited Supervision. European Conference on Computer Vision (ECCV) 2024.
1. [ET] Zeng et al. DUAL: deep unsupervised simultaneous simulation and denoising for cryo-electron tomography. bioRxiv. [doi:10.1101/2024.03.02.583135](https://doi.org/10.1101/2024.03.02.583135)
1. [CV] Wu et al. One-Prompt to Segment All Medical Images. IEEE conference on computer vision and pattern recognition (CVPR) 2024. [doi:10.48550/arXiv.2305.10300](https://doi.org/10.48550/arXiv.2305.10300)
1. [ET] Zeng et al. High-throughput cryo-ET structural pattern mining by deep unsupervised clustering. Proceedings of the National Academy of Sciences. [doi:10.1073/pnas.2213149120](https://www.pnas.org/doi/10.1073/pnas.2213149120)
1. [AS,CV] Sarker et al. Detecting Anomalies from Liquid Transfer Videos in Automated Laboratory Setting. Frontiers in Molecular Biosciences. [doi:10.3389/fmolb.2023.1147514](https://doi.org/10.3389/fmolb.2023.1147514)
1. [CV,ML,ET,EM] Uddin et al. Harmony: A Generic Unsupervised Approach for Disentangling Semantic Content from Parameterized Transformations. IEEE conference on computer vision and pattern recognition (CVPR 2022). [Paper](https://openaccess.thecvf.com/content/CVPR2022/html/Uddin_Harmony_A_Generic_Unsupervised_Approach_for_Disentangling_Semantic_Content_From_CVPR_2022_paper.html)
1. [ML] Wang et al. Boosting Active Learning via Improving Test Performance. AAAI Conference on Artificial Intelligence (AAAI) 2022. [arXiv:2112.05683](https://arxiv.org/abs/2112.05683)
1. [CV,ET] Zeng et al. End-to-end robust joint unsupervised image alignment and clustering. International Conference on Computer Vision (ICCV 2021). [doi:10.1109/iccv48922.2021.00383](https://doi.org/10.1109/iccv48922.2021.00383)
1. [CV,ET] Zhu et al. Weakly Supervised 3D Semantic Segmentation Using Cross-Image Consensus and Inter-Voxel Affinity Relations. International Conference on Computer Vision (ICCV 2021). [doi:10.1109/icv48922.2021.00283](https://doi.org/10.1109/iccv48922.2021.00283)
1. [ET] Du et al. Active learning to classify macromolecular structures in situ for less supervision in cryo-electron tomography. Bioinformatics. [doi:10.1093/bioinformatics/btab123](https://doi.org/10.1093/bioinformatics/btab123) [arXiv:2102.12040](https://arxiv.org/abs/2102.12040)
1. [ET] Zeng X, Xu M. Gum-Net: Unsupervised geometric matching for fast and accurate 3D subtomogram image alignment and averaging. In Proceedings of the IEEE conference on computer vision and pattern recognition (CVPR 2020). [Paper](http://openaccess.thecvf.com/content_CVPR_2020/html/Zeng_Gum-Net_Unsupervised_Geometric_Matching_for_Fast_and_Accurate_3D_Subtomogram_CVPR_2020_paper.html)
1. [ET] Li et al. Few-shot learning for classification of novel macromolecular structures in cryo-electron tomograms. PLOS Computational Biology. [doi:10.1371/journal.pcbi.1008227](https://doi.org/10.1371/journal.pcbi.1008227)
1. [ET] Du et al. Open-set Recognition of Unseen Macromolecules in Cellular Electron Cryo-Tomograms by Soft Large Margin Centralized Cosine Loss. British Machine Vision Conference (BMVC 2019 spotlight with acceptance rate < 5%). [pdf](https://www.google.com/url?q=https%3A%2F%2Fbmvc2019.org%2Fwp-content%2Fuploads%2Fpapers%2F0347-paper.pdf&sa=D&sntz=1&usg=AFQjCNHsPGgSr-V9nd4XQgRo6XrbD-Gw3g)
1. [ET] Zhao et al. Respond-CAM: Analyzing Deep Models for 3D Imaging Data by Visualizations. Medical Image Computing & Computer Assisted Intervention (MICCAI) 2018. [arXiv:1806.00102](https://www.google.com/url?q=https%3A%2F%2Farxiv.org%2Fabs%2F1806.00102&sa=D&sntz=1&usg=AFQjCNGgP4R3YTn18pWjf3tItO1EJcBocg)
1. [ET] Zhao et al. An integration of fast alignment and maximum-likelihood methods for electron subtomogram averaging and classification. ISMB 2018. Bioinformatics. 2018 Jul 1; 34(13): i227â€“i236\. [doi:10.1093/bioinformatics/bty267](https://www.google.com/url?q=https%3A%2F%2Fdoi.org%2F10.1093%2Fbioinformatics%2Fbty267&sa=D&sntz=1&usg=AFQjCNEzaiJD5hLMLjD4MIKW2O9gJ-Icvg). [arXiv:1804.01203](https://www.google.com/url?q=https%3A%2F%2Farxiv.org%2Fabs%2F1804.01203&sa=D&sntz=1&usg=AFQjCNFUZrHiXv_yh0KWf1DO4qAy1IHOWQ)
1. [ET] Zeng et al. A convolutional autoencoder approach for mining features in cellular electron cryo-tomograms and weakly supervised coarse segmentation. Journal of Structural Biology. 2018 May;202(2):150-160\. [doi:10.1016/j.jsb.2017.12.015](https://www.google.com/url?q=https%3A%2F%2Fdoi.org%2F10.1016%2Fj.jsb.2017.12.015&sa=D&sntz=1&usg=AFQjCNHd6jJBB-Obp_ebGPvYCwwXv9UrkA) [arXiv:1706.04970](https://www.google.com/url?q=https%3A%2F%2Farxiv.org%2Fabs%2F1706.04970&sa=D&sntz=1&usg=AFQjCNFez1S5-gvM5whMCXe1yU5-vXVa2g) [[code](https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fxulabs%2Fprojects%2Ftree%2Fmaster%2Fautoencoder&sa=D&sntz=1&usg=AFQjCNG3kvV1s4UAW8nIBaqZN67rbqN-uA)]
1. [ET] Xu et al. De novo structural pattern mining in cellular electron cryo-tomograms. Structure. 2019 Apr 2;27(4):679-691.e14\. [doi:10.1016/j.str.2019.01.005](https://doi.org/10.1016/j.str.2019.01.005). (Appeared on Structure volume cover and highlighted in Nature Methods 16, page 285 (2019), [doi:10.1038/s41592-019-0382-2](https://www.google.com/url?q=https%3A%2F%2Fdoi.org%2F10.1038%2Fs41592-019-0382-2&sa=D&sntz=1&usg=AFQjCNHMCQyejm_T4RGn8t5EKPadA4XFzA))
1. [OM] Wang et al. Learn to segment single cells with deep distance estimator and deep cell detector. Computers in Biology and Medicine. [doi:10.1016/j.compbiomed.2019.04.006](https://doi.org/10.1016/j.compbiomed.2019.04.006)
1. [ET] Xu et al. High-throughput subtomogram alignment and classification by Fourier space constrained fast volumetric matching. Journal of Structural Biology. 2012 May;178(2):152-64\. Epub 2012 Mar 7.
1. [GE] Xu et al. Automated multidimensional phenotype profiling using large public microarray repositories. Proc Natl Acad Sci U S A. (PNAS). 2009; 106(30), 12323 - 12328\. (Highlighted in Nature Methods 6, 632; Selected and re-published in 2010 International Medical Informatics Association Yearbook of Medical Informatics)
1. [GE] Xu et al. An integrative approach to characterize disease-specific pathways and their coordination: a case study in cancer. BMC Genomics. 2008; 9 Suppl 1:S12.

Categories:

* ET: Cryo-Electron Tomography
* EM: Cryo-Electron Microscopy
* AS: Automated Science
* CV: Computer Vision
* ML: Machine Learning
* OM: Optical Microscopy
* GE: Computational Genomics
