<!doctype html>
<!--
  Minimal Mistakes Jekyll Theme 4.22.0 by Michael Rose
  Copyright 2013-2020 Michael Rose - mademistakes.com | @mmistakes
  Free for personal and commercial use under the MIT license
  https://github.com/mmistakes/minimal-mistakes/blob/master/LICENSE
-->
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">

<!-- begin _includes/seo.html --><title>Software Tutorial: Visualizing Specific Regions of Interest within a Protein Structure - Xu lab@CBD_SCS_CMU</title>
<meta name="description" content="Write an awesome description for your new site here. You can edit this line in _config.yml. It will appear in your document head meta (for Google search results) and in your feed.xml site description.">



<meta property="og:type" content="website">
<meta property="og:locale" content="en_US">
<meta property="og:site_name" content="Xu lab@CBD_SCS_CMU">
<meta property="og:title" content="Software Tutorial: Visualizing Specific Regions of Interest within a Protein Structure">
<meta property="og:url" content="http://localhost:4000/coronavirus/tutorial_visualization">


  <meta property="og:description" content="Write an awesome description for your new site here. You can edit this line in _config.yml. It will appear in your document head meta (for Google search results) and in your feed.xml site description.">












<link rel="canonical" href="http://localhost:4000/coronavirus/tutorial_visualization">




<script type="application/ld+json">
  {
    "@context": "https://schema.org",
    
      "@type": "Person",
      "name": null,
      "url": "http://localhost:4000/"
    
  }
</script>






<!-- end _includes/seo.html -->



<!-- https://t.co/dKP3o1e -->
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<script>
  document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/g, '') + ' js ';
</script>

<!-- For all browsers -->
<link rel="stylesheet" href="/assets/css/main.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5/css/all.min.css">

<!--[if IE]>
  <style>
    /* old IE unsupported flexbox fixes */
    .greedy-nav .site-title {
      padding-right: 3em;
    }
    .greedy-nav button {
      position: absolute;
      top: 0;
      right: 0;
      height: 100%;
    }
  </style>
<![endif]-->


    <!-- start custom head snippets -->

<!-- insert favicons. use https://realfavicongenerator.net/ -->

<!-- end custom head snippets -->

  </head>

  <body class="layout--single">
    <nav class="skip-links">
  <h2 class="screen-reader-text">Skip links</h2>
  <ul>
    <li><a href="#site-nav" class="screen-reader-shortcut">Skip to primary navigation</a></li>
    <li><a href="#main" class="screen-reader-shortcut">Skip to content</a></li>
    <li><a href="#footer" class="screen-reader-shortcut">Skip to footer</a></li>
  </ul>
</nav>

    <!--[if lt IE 9]>
<div class="notice--danger align-center" style="margin: 0;">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience.</div>
<![endif]-->

    <div class="masthead">
  <div class="masthead__inner-wrap">
    <div class="masthead__menu">
      <nav id="site-nav" class="greedy-nav">
        <a class="site-title" href="/">Xu lab@CBD_SCS_CMU</a>
        <ul class="visible-links">
                
                
                    <li class="masthead__menu-item">
                        <a href="/news/">NEWS</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/papers/">PAPERS</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/group/">GROUP</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/meet-the-team/">ABOUT MIN XU</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/position/">POSITION</a>
                    </li>
                
            
                
                
                    <li class="masthead__menu-item">
                        <a href="/software/">SOFTWARE</a>
                    </li>
                
            
                
                
                    <li class="dropdown ">
                        <a class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">Take the Course    <i class="fa fa-caret-down fa-sm" aria-hidden="true"></i><span class="caret"></span></a>
                        <ul class="dropdown-content">
                            
                                <li>
                                    <a href="/prologue/">    <b>Prologue:</b> Introducing biological modeling with Turing patterns</a>
                                </li>
                            
                                <li>
                                    <a href="/motifs/home">    <b>Module 1:</b> Finding motifs in transcription factor networks</a>
                                </li>
                            
                                <li>
                                    <a href="/chemotaxis/home">    <b>Module 2:</b> Unpacking E. coli's genius exploration algorithm</a>
                                </li>
                            
                                <li>
                                    <a href="/coronavirus/home">    <b>Module 3:</b> Analyzing the coronavirus spike protein</a>
                                </li>
                            
                                <li>
                                    <a href="">    <b>Module 4:</b> coming soon!</a>
                                </li>
                            
                        </ul>
                    </li>
                
            
        </ul>
        <!--
        
        <button class="search__toggle" type="button">
          <svg class="icon" width="16" height="16" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 15.99 16">
            <path d="M15.5,13.12L13.19,10.8a1.69,1.69,0,0,0-1.28-.55l-0.06-.06A6.5,6.5,0,0,0,5.77,0,6.5,6.5,0,0,0,2.46,11.59a6.47,6.47,0,0,0,7.74.26l0.05,0.05a1.65,1.65,0,0,0,.5,1.24l2.38,2.38A1.68,1.68,0,0,0,15.5,13.12ZM6.4,2A4.41,4.41,0,1,1,2,6.4,4.43,4.43,0,0,1,6.4,2Z" transform="translate(-.01)"></path>
          </svg>
        </button>
        
        -->
        <button class="greedy-nav__toggle hidden" type="button">
          <span class="visually-hidden">Toggle Menu</span>
          <div class="navicon"></div>
        </button>
        <ul class="hidden-links hidden"></ul>
      </nav>
    </div>
  </div>
</div>


    <div class="initial-content">
      



<div id="main" role="main">

  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>

  
  <div class="sidebar sticky">
  
  
    
      
      
      
      
    
    
      

<nav class="nav__list">
  
  <input id="ac-toc" name="accordion-toc" type="checkbox" />
  <label for="ac-toc">Toggle Menu</label>
  <ul class="nav__items">
    
      <li>
        
          <span class="nav__sub-title">Introduction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/home">A Tale of Two Doctors<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 1: Protein Structure Prediction</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/structure_intro">An Introduction to Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/ab_initio">Ab initio Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/homology">Homology Modeling for Protein Structure Prediction<br><br></a></li>
          
            <li><a href="/coronavirus/accuracy">Comparing Protein Structures to Assess Model Accuracy<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_1">Part 1 Conclusion: Protein Structure Prediction is Solved! (Kinda)<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Part 2: Comparing SARS-CoV-2 and SARS</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/multiseq">Searching for Local Differences in the SARS-CoV and SARS-CoV-2 Spike Proteins<br><br></a></li>
          
            <li><a href="/coronavirus/structural_diff">Analyzing Structural Differences in the Bonding of SARS-CoV and SARS-CoV-2 with the ACE2 Enzyme<br><br></a></li>
          
            <li><a href="/coronavirus/NAMD">Quantifying the Interaction Energy Between the SARS-CoV-2 Spike Protein and ACE2<br><br></a></li>
          
            <li><a href="/coronavirus/conclusion_part_2">Part 2 Conclusion: From Static Protein Analysis to Molecular Dynamics<br><br></a></li>
          
        </ul>
        
      </li>
    
      <li>
        
          <span class="nav__sub-title">Software Tutorials (featuring ProDy)</span>
        

        
        <ul>
          
            <li><a href="/coronavirus/tutorial_ab_initio">Using ab initio Modeling to Predict the Structure of Hemoglobin Subunit Alpha<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_homology">Using Homology Modeling to Predict the Structure of the SARS-CoV-2 Spike Protein<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_rmsd">Using RMSD to Compare Two Protein Structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_multiseq">Finding Local Differences in Two Protein Structures<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_visualization" class="active">Visualizing Specific Regions of Interest within a Protein Structure<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_NAMD">Computing the Energy Contributed by a Local Region of a Bound Complex<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_GNM">Molecular Dynamics Analysis with GNM<br><br></a></li>
          
            <li><a href="/coronavirus/tutorial_ANM">ANM Simulations Add Directionality to GNM<br><br></a></li>
          
        </ul>
        
      </li>
    
  </ul>
</nav>

    
  
  </div>



  <article class="page" itemscope itemtype="https://schema.org/CreativeWork">
    <meta itemprop="headline" content="Software Tutorial: Visualizing Specific Regions of Interest within a Protein Structure">
    
    
    

    <div class="page__inner-wrap">
      
        <header>
          <h1 id="page-title" class="page__title" itemprop="headline">Software Tutorial: Visualizing Specific Regions of Interest within a Protein Structure
</h1>
          
        </header>
      

      <section class="page__content" itemprop="text">
        
          <aside class="sidebar__right sticky">
            <nav class="toc">
              <header><h4 class="nav__title"><i class="fas fa-file-alt"></i> On this page</h4></header>
              

            </nav>
          </aside>
        
        <p>In this tutorial, we will discuss how to visualize a protein structure and highlight specific amino acids of interest in the protein. We will focus on the region that we identified in the previous tutorial starting at around position 475 of the SARS-CoV-2 RBD, where we found that this RBD differs structurally from that of SARS-CoV.</p>

<p>We will visualize the site in the SARS-CoV-2 RBD using the SARS-CoV-2 chimeric RBD complexed with the human ACE2 enzyme (PDB entry: <a href="https://www.rcsb.org/structure/6vw1" target="_blank">6vw1</a>). Before completing this tutorial, you should have installed VMD and know how to load molecules into the program. If you need a refresher, visit the <a href="tutorial_multiseq" target="_blank">previous tutorial</a>.</p>

<p>First, download the chimeric RBD <code class="language-plaintext highlighter-rouge">.pdb</code> file and load it into VMD.</p>

<p>To change the visualization of the protein, click <code class="language-plaintext highlighter-rouge">Graphics &gt; Representation</code>. Double clicking on a representation will enable/disable it.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge4.png" alt="image-center" class="align-center" /></p>

<p>The file <code class="language-plaintext highlighter-rouge">6vw1.pdb</code> contains two biological assemblies of the complex. The first assembly contains Chain A (taken from ACE2) and chain E (taken from RBD), and the second assembly contains chain B (ACE2) and chain F (RBD). We will focus on the second assembly.</p>

<p>We will first add chain B, from the ACE2 enzyme, and color it green.</p>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">Selected Atoms</code> allows you to select specific parts of the molecule. The keyword <code class="language-plaintext highlighter-rouge">all</code> selects all atoms in the file, and so replace <code class="language-plaintext highlighter-rouge">all</code> with <code class="language-plaintext highlighter-rouge">chain B</code>. Then, click on <code class="language-plaintext highlighter-rouge">Apply</code>. (In general, to choose a specific chain, use the expression <code class="language-plaintext highlighter-rouge">chain X</code>, where <code class="language-plaintext highlighter-rouge">X</code> is the chain of interest. To choose a specific residue (i.e., amino acid), use the keyword <code class="language-plaintext highlighter-rouge">resid #</code>. Expressions can be combined using the keywords <code class="language-plaintext highlighter-rouge">and</code> and <code class="language-plaintext highlighter-rouge">or</code>, and more complicated selections need parentheses.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">Coloring Method</code> allows you to change the coloring method of the selected atoms. This includes coloring based on element, amino acid residue, and many more. To choose a specific color, select <code class="language-plaintext highlighter-rouge">ColorID</code>. A drop-down list will appear to color selection. Choose “7” to select green.</p>
  </li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">Drawing Method</code> allows you to change the visualization of the selected atoms. <code class="language-plaintext highlighter-rouge">Lines</code> (also known as <code class="language-plaintext highlighter-rouge">wireframe</code>) draws a line between atoms to represent bonds. <code class="language-plaintext highlighter-rouge">Tube</code> focuses only on the backbone of the molecule.  <code class="language-plaintext highlighter-rouge">Licorice</code> will show both the backbone and the side chains of the protein. <code class="language-plaintext highlighter-rouge">Cartoon/NewCartoon</code> will show the secondary structure of the molecule (protein). We are interesed mostly in the backbone, and so we will choose <code class="language-plaintext highlighter-rouge">Tube</code>.</p>
  </li>
</ul>

<p style="font-size: medium;"><img src="../assets/images/Ridge5.png" alt="image-center" class="align-center" /></p>

<p>At this point, your <code class="language-plaintext highlighter-rouge">OpenGL Display</code> window should look like the following:</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge6.png" alt="image-center" class="align-center" /></p>

<p>We next add chain F, from the SARS-CoV-2 chimeric RBD, and color it purple. Click on <code class="language-plaintext highlighter-rouge">Create Rep</code>, which will duplicate the previous representation. Then, change <code class="language-plaintext highlighter-rouge">Selected Atoms</code> to <code class="language-plaintext highlighter-rouge">chain F</code> and <code class="language-plaintext highlighter-rouge">ColoringID</code> to “11” to color the chain purple. Make sure your other selections are as follows:</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge7.png" alt="image-center" class="align-center" /></p>

<p>You should now see two distinct colored structures!</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge8.png" alt="image-center" class="align-center" /></p>

<p>We can also change our visualization to target specific amino acids by creating another representation and specifying the amino acid with the keyword <code class="language-plaintext highlighter-rouge">resid</code> followed by the position of this amino acid residue.</p>

<p>For example, say that we are interested in residue 486 in the RBD (which is phenylalanine). Click on <code class="language-plaintext highlighter-rouge">Create Rep</code>. In the new representation, change <code class="language-plaintext highlighter-rouge">Selected Atoms</code> to <code class="language-plaintext highlighter-rouge">chain F and resid 486</code> and click <code class="language-plaintext highlighter-rouge">Apply</code>. Then change the <code class="language-plaintext highlighter-rouge">Coloring Method</code> to <code class="language-plaintext highlighter-rouge">ColorID</code> and <code class="language-plaintext highlighter-rouge">4</code>. Finally, change the <code class="language-plaintext highlighter-rouge">Drawing Method</code> to <code class="language-plaintext highlighter-rouge">Licorice</code>.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge8-1.png" alt="image-center" class="align-center" /></p>

<p>In the <code class="language-plaintext highlighter-rouge">OpenGL Display</code> window, you will now see a new yellow projection coming out of the RBD, as shown in the image below. This is residue 486! You may need to rotate the protein a bit to see it. (Instructions on how to rotate a molecule and zoom in and out within VMD were given in our <a href="tutorial_multiseq">tutorial on finding local protein differences</a>.)</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge8-2.png" alt="image-center" class="align-center" /></p>

<p>Let’s now color a few more residues from our region of interest: residues 475 and 487 of the RBD, and residues 19, 24, 79, 82, and 83 of ACE2. As we return to the main text, we will explain why these residues are implicated in binding affinity.</p>

<p>Coloring these residues is analogous to the previous steps of just adding new representations and changing <code class="language-plaintext highlighter-rouge">Selected Atoms</code>, <code class="language-plaintext highlighter-rouge">Coloring Method</code>, and <code class="language-plaintext highlighter-rouge">Drawing Method</code>. Make the following representations; note the colors that we use.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge9.png" alt="image-center" class="align-center" /></p>

<p>Your final visualization should look like the following figure.</p>

<p style="font-size: medium;"><img src="../assets/images/Ridge10.png" alt="image-center" class="align-center" /></p>

<p>Congratulations! You have now created a detailed visualization of the RBD-ACE2 complex that focuses on our site of interest. As we return to the main text, we will discuss how the highlighted amino acids help increase the binding affinity of the SARS-CoV-2 spike protein to ACE2.</p>

<p><strong>STOP:</strong> Create another visualization of the same site using the SARS-CoV RBD complex with ACE2 (PDB entry: <a href="https://www.rcsb.org/structure/2AJF" target="_blank">2ajf</a>). How does it compare with your first visualization of SARS-CoV-2 complex? Use the following graphical representations:</p>

<p>SARS-CoV RBD with ACE2 (2ajf)</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Protein</th>
      <th style="text-align: left">Style</th>
      <th style="text-align: left">Color</th>
      <th style="text-align: left">Selection</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">SARS-CoV RBD</td>
      <td style="text-align: left">Tube</td>
      <td style="text-align: left">ColorID 11</td>
      <td style="text-align: left">chain F</td>
    </tr>
    <tr>
      <td style="text-align: left">SARS-CoV RBD</td>
      <td style="text-align: left">Licorice</td>
      <td style="text-align: left">ColorID 4</td>
      <td style="text-align: left">chain F and resid 472</td>
    </tr>
    <tr>
      <td style="text-align: left">ACE2</td>
      <td style="text-align: left">Licorice</td>
      <td style="text-align: left">ColorID 6</td>
      <td style="text-align: left">chain B and (resid 82 or resid 79 or resid 83)</td>
    </tr>
    <tr>
      <td style="text-align: left">ACE2</td>
      <td style="text-align: left">Licorice</td>
      <td style="text-align: left">ColorID 10</td>
      <td style="text-align: left">chain B and resid 19</td>
    </tr>
    <tr>
      <td style="text-align: left">ACE2</td>
      <td style="text-align: left">Licorice</td>
      <td style="text-align: left">ColorID 3</td>
      <td style="text-align: left">chain B and resid 24</td>
    </tr>
  </tbody>
</table>

<p style="font-size: 100%; text-align: center;"><a href="structural_diff#analyzing-three-sites-of-conformational-differences" class="btn btn--primary btn--large">Return to main text</a></p>

        
      </section>

      <footer class="page__meta">
        
        


        

      </footer>

      

      
    </div>

    
  </article>

  <!-- only show related on a post page when `related: true`
  
  
  -->
</div>

    </div>

    
      <div class="search-content">
        <div class="search-content__inner-wrap"><form class="search-content__form" onkeydown="return event.key != 'Enter';">
    <label class="sr-only" for="search">
      Enter your search term...
    </label>
    <input type="search" id="search" class="search-input" tabindex="-1" placeholder="Enter your search term..." />
  </form>
  <div id="results" class="results"></div></div>

      </div>
    

    <div id="footer" class="page__footer">
      <footer>
        <!-- start custom footer snippets -->

<!-- end custom footer snippets -->
        <div class="page__footer-follow">
  <hr>
  <a href="http://cbd.cmu.edu" target="_blank"><img align = "left" src="/biological_modeling_course/assets/images/combined_SCS_Unitmark_187_K.jpg" alt="CBD Logo" width="40%" height="40%"></a>
  <a href="http://mmbios.pitt.edu" target="_blank"><img align="right" src="/biological_modeling_course/assets/images/MMBioS_Logo.png" alt="MMBioS Logo" width="50%" height="50%"></a>
</div>

      </footer>
    </div>

    
  <script src="/assets/js/main.min.js"></script>




<script src="/assets/js/lunr/lunr.min.js"></script>
<script src="/assets/js/lunr/lunr-store.js"></script>
<script src="/assets/js/lunr/lunr-en.js"></script>







  </body>
</html>
